test/metrics/metrics_test.go

```diff
@@ -0,0 +1,735 @@
+
+			// when deleting the host-operator pod to emulate an operator restart during redeployment.
+			err := hostAwait.DeletePods(client.InNamespace(hostAwait.Namespace), client.MatchingLabels{"name": "controller-manager"})
+
+			// then check how much time it takes to restart and process all existing resources
+			require.NoError(t, err)
+
+			// host metrics should become available again at this point
+			_, err = hostAwait.WaitForRouteToBeAvailable(hostAwait.Namespace, "host-operator-metrics-service", "/metrics")
+			require.NoError(t, err, "failed while setting up or waiting for the route to the 'host-operator-metrics-service' service to be available")
+
+			// then verify that the metric values "survived" the restart
+			metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 0, "activations", "1", "domain", "external") // user-0001 was 1 time (unchanged after pod restarted)
+			metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 0, "activations", "2", "domain", "external") // user-0002 was 2 times (unchanged after pod restarted)
+			metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 0, "activations", "3", "domain", "external") // user-0003 was 3 times (unchanged after pod restarted)
+		})
+
+		t.Run("delete usersignups", func(t *testing.T) {
+			// given
+			metricsAssertion := InitMetricsAssertion(t, hostAwait, []string{memberAwait.ClusterName, member2Await.ClusterName})
+
+			// when deleting user "user-0001"
+			err := hostAwait.Client.Delete(context.TODO(), usersignups["user-0001"])
+
+			// then
+			require.NoError(t, err)
+			// and verify that the values of the `sandbox_users_per_activations` metric
+			metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 0, "activations", "1", "domain", "external") // user-0001 has been deleted but metric remains unchanged
+			metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 0, "activations", "2", "domain", "external") // (unchanged after other usersignup was deleted)
+			metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 0, "activations", "3", "domain", "external") // (unchanged after other usersignup was deleted)
+
+			// when deleting user "user-0002"
+			err = hostAwait.Client.Delete(context.TODO(), usersignups["user-0002"])
+
+			// then
+			require.NoError(t, err)
+			// and verify that the values of the `sandbox_users_per_activations` metric
+			metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 0, "activations", "1", "domain", "external") // (same offset as above)
+			metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 0, "activations", "2", "domain", "external") // user-0002 has been deleted but metric remains unchanged
+			metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 0, "activations", "3", "domain", "external") // (unchanged after other usersignup was deleted)
+
```

Comment:
having these two cases `restart host-operator pod and verify that metrics are still available` & `delete usersignups` as sub-tests of `when users reactivated` is a bit confusing - they don't have anything in common with reactivation - they only reuse the users